finagle-core/src/main/scala/com/twitter/finagle/tracing/Trace.scala

```diff
@@ -143,9 +148,9 @@ object Trace {
     val spanId = SpanId(rng.nextLong())
     idOption match {
       case Some(id) =>
-        TraceId(Some(id.traceId), Some(id.spanId), spanId, id.sampled, id.flags)
+        TraceId(Some(id.traceId), Some(id.spanId), spanId, id.sampled, id.flags, id.traceIdHigh)
```

Comment:
An aside: this is so close to being `id.copy(spanId = spanId)` but I'm worried about the subtle differences in `def traceId` etc.